struct.c

```diff
@@ -510,65 +509,123 @@ rb_struct_define_under(VALUE outer, const char *name, ...)
- *  used to create instances of the structure like any other Class.
- *
- *  If the +class_name+ is omitted an anonymous structure class will be
- *  created.  Otherwise, the name of this struct will appear as a constant in
- *  class Struct, so it must be unique for all Structs in the system and
- *  must start with a capital letter.  Assigning a structure class to a
- *  constant also gives the class the name of the constant.
- *
- *     # Create a structure with a name under Struct
- *     Struct.new("Customer", :name, :address)
- *     #=> Struct::Customer
- *     Struct::Customer.new("Dave", "123 Main")
- *     #=> #<struct Struct::Customer name="Dave", address="123 Main">
- *
- *     # Create a structure named by its constant
- *     Customer = Struct.new(:name, :address)
- *     #=> Customer
- *     Customer.new("Dave", "123 Main")
- *     #=> #<struct Customer name="Dave", address="123 Main">
- *
- *  If the optional +keyword_init+ keyword argument is set to +true+,
- *  .new takes keyword arguments instead of normal arguments.
- *
- *     Customer = Struct.new(:name, :address, keyword_init: true)
- *     Customer.new(name: "Dave", address: "123 Main")
- *     #=> #<struct Customer name="Dave", address="123 Main">
- *
- *  If a block is given it will be evaluated in the context of
- *  +StructClass+, passing the created class as a parameter:
+ *    Struct.new(class_name) -> new_class
+ *    Struct.new(*member_names, **kwargs) -> new_class
+ *    Struct.new(class_name, *member_names, **kwargs) -> new_class
+ *    Struct.new(class_name) {|new_class| ... } -> new_class
+ *    Struct.new(*member_names, **kwargs) {|new_class| ... } -> new_class
+ *    Struct.new(class_name, *member_names, **kwargs) {|new_class| ... } -> new_class
+ *
+ *  Returns a new subclass of +Struct+.  The new subclass:
+ *
+ *  - May be anonymous, or may have the name given by +class_name+.
+ *  - May have members as given by +member_names+.
+ *  - May have initialization via ordinary arguments (the default)
+ *    or via keyword arguments (if <tt>keyword_init: true</tt> is given).
+ *
+ *  <b>\Class Name</b>
+ *
+ *  With string argument +class_name+,
+ *  returns a new subclass of +Struct+ named <tt>Struct::<em>class_name</em></tt>:
+ *
+ *    # Class name given, but no member names.
+ *    Foo = Struct.new('Foo')             # => Struct::Foo
+ *    Foo.name                            # => "Struct::Foo"
+ *    Foo.superclass                      # => Struct
+ *    # Class name and member names given.
+ *    Foo = Struct.new('Foo', :foo, :bar) # => Struct::Foo
+ *    Foo.name                            # => "Struct::Foo"
+ *    Foo.superclass                      # => Struct
+ *
+ *  Without string argument +class_name+,
+ *  returns a new anonymous subclass of +Struct+:
+ *
+ *    Struct.new(:foo, :bar).name # => nil
+ *
+ *  <b>Member Names</b>
+ *
+ *  With symbol arguments +member_names+,
+ *  returns a new subclass of +Struct+ with the given members:
```

Comment:
`member_names` is almost always going to be provided, since a Struct with no members is pretty useless.  So this should say that the `member_names` are used to set the members in the returned Struct subclass.  I wouldn't even mention usage without member names.

Reply:
Changed.